File: src/nested_switch.rs
pub fn nested_switch(input: &str) -> Option<usize> {
    match input.len() {
        // if, in, do
        2 => {
            match input.as_bytes()[0] as char {
                // do
                'd' => {
                    if input.as_bytes()[1] as char == 'o' {
                        return Some(40);
                    } else {
                        return None;
                    }
                }

                // if, in
                'i' => {
                    match input.as_bytes()[1] as char {
                        // if
                        'f' => {
                            return Some(1);
                        }

                        // in
                        'n' => {
                            return Some(24);
                        }

                        _ => {
                            return None;
                        }
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // var, get, for, new, any, let, try
        3 => {
            match input.as_bytes()[0] as char {
                // any
                'a' => {
                    if (input.as_bytes()[1] as char == 'n') & (input.as_bytes()[2] as char == 'y') {
                        return Some(28);
                    } else {
                        return None;
                    }
                }

                // for
                'f' => {
                    if (input.as_bytes()[1] as char == 'o') & (input.as_bytes()[2] as char == 'r') {
                        return Some(17);
                    } else {
                        return None;
                    }
                }

                // get
                'g' => {
                    if (input.as_bytes()[1] as char == 'e') & (input.as_bytes()[2] as char == 't') {
                        return Some(7);
                    } else {
                        return None;
                    }
                }

                // let
                'l' => {
                    if (input.as_bytes()[1] as char == 'e') & (input.as_bytes()[2] as char == 't') {
                        return Some(31);
                    } else {
                        return None;
                    }
                }

                // new
                'n' => {
                    if (input.as_bytes()[1] as char == 'e') & (input.as_bytes()[2] as char == 'w') {
                        return Some(23);
                    } else {
                        return None;
                    }
                }

                // try
                't' => {
                    if (input.as_bytes()[1] as char == 'r') & (input.as_bytes()[2] as char == 'y') {
                        return Some(36);
                    } else {
                        return None;
                    }
                }

                // var
                'v' => {
                    if (input.as_bytes()[1] as char == 'a') & (input.as_bytes()[2] as char == 'r') {
                        return Some(4);
                    } else {
                        return None;
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // case, else, type, enum, void, null, this, true
        4 => {
            match input.as_bytes()[0] as char {
                // case
                'c' => {
                    if (input.as_bytes()[1] as char == 'a')
                        & ((input.as_bytes()[2] as char == 's')
                            & (input.as_bytes()[3] as char == 'e'))
                    {
                        return Some(0);
                    } else {
                        return None;
                    }
                }

                // else, enum
                'e' => {
                    match input.as_bytes()[1] as char {
                        // else
                        'l' => {
                            if (input.as_bytes()[2] as char == 's')
                                & (input.as_bytes()[3] as char == 'e')
                            {
                                return Some(3);
                            } else {
                                return None;
                            }
                        }

                        // enum
                        'n' => {
                            if (input.as_bytes()[2] as char == 'u')
                                & (input.as_bytes()[3] as char == 'm')
                            {
                                return Some(14);
                            } else {
                                return None;
                            }
                        }

                        _ => {
                            return None;
                        }
                    }
                }

                // null
                'n' => {
                    if (input.as_bytes()[1] as char == 'u')
                        & ((input.as_bytes()[2] as char == 'l')
                            & (input.as_bytes()[3] as char == 'l'))
                    {
                        return Some(20);
                    } else {
                        return None;
                    }
                }

                // type, this, true
                't' => {
                    match input.as_bytes()[1] as char {
                        // this
                        'h' => {
                            if (input.as_bytes()[2] as char == 'i')
                                & (input.as_bytes()[3] as char == 's')
                            {
                                return Some(22);
                            } else {
                                return None;
                            }
                        }

                        // true
                        'r' => {
                            if (input.as_bytes()[2] as char == 'u')
                                & (input.as_bytes()[3] as char == 'e')
                            {
                                return Some(26);
                            } else {
                                return None;
                            }
                        }

                        // type
                        'y' => {
                            if (input.as_bytes()[2] as char == 'p')
                                & (input.as_bytes()[3] as char == 'e')
                            {
                                return Some(9);
                            } else {
                                return None;
                            }
                        }

                        _ => {
                            return None;
                        }
                    }
                }

                // void
                'v' => {
                    if (input.as_bytes()[1] as char == 'o')
                        & ((input.as_bytes()[2] as char == 'i')
                            & (input.as_bytes()[3] as char == 'd'))
                    {
                        return Some(19);
                    } else {
                        return None;
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // throw, while, super, false, yield, const, catch
        5 => {
            match input.as_bytes()[0] as char {
                // const, catch
                'c' => {
                    match input.as_bytes()[1] as char {
                        // catch
                        'a' => {
                            if (input.as_bytes()[2] as char == 't')
                                & ((input.as_bytes()[3] as char == 'c')
                                    & (input.as_bytes()[4] as char == 'h'))
                            {
                                return Some(41);
                            } else {
                                return None;
                            }
                        }

                        // const
                        'o' => {
                            if (input.as_bytes()[2] as char == 'n')
                                & ((input.as_bytes()[3] as char == 's')
                                    & (input.as_bytes()[4] as char == 't'))
                            {
                                return Some(38);
                            } else {
                                return None;
                            }
                        }

                        _ => {
                            return None;
                        }
                    }
                }

                // false
                'f' => {
                    if input == "false" {
                        return Some(27);
                    } else {
                        return None;
                    }
                }

                // super
                's' => {
                    if input == "super" {
                        return Some(21);
                    } else {
                        return None;
                    }
                }

                // throw
                't' => {
                    if input == "throw" {
                        return Some(2);
                    } else {
                        return None;
                    }
                }

                // while
                'w' => {
                    if input == "while" {
                        return Some(18);
                    } else {
                        return None;
                    }
                }

                // yield
                'y' => {
                    if input == "yield" {
                        return Some(37);
                    } else {
                        return None;
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // number, string, module, typeof, public, export, return, static
        6 => {
            match input.as_bytes()[0] as char {
                // export
                'e' => {
                    if input == "export" {
                        return Some(15);
                    } else {
                        return None;
                    }
                }

                // module
                'm' => {
                    if input == "module" {
                        return Some(8);
                    } else {
                        return None;
                    }
                }

                // number
                'n' => {
                    if input == "number" {
                        return Some(5);
                    } else {
                        return None;
                    }
                }

                // public
                'p' => {
                    if input == "public" {
                        return Some(12);
                    } else {
                        return None;
                    }
                }

                // return
                'r' => {
                    if input == "return" {
                        return Some(25);
                    } else {
                        return None;
                    }
                }

                // string, static
                's' => {
                    match input.as_bytes()[2] as char {
                        // static
                        'a' => {
                            if (input.as_bytes()[3] as char == 't')
                                & ((input.as_bytes()[4] as char == 'i')
                                    & (input.as_bytes()[5] as char == 'c'))
                            {
                                return Some(30);
                            } else {
                                return None;
                            }
                        }

                        // string
                        'r' => {
                            if (input.as_bytes()[3] as char == 'i')
                                & ((input.as_bytes()[4] as char == 'n')
                                    & (input.as_bytes()[5] as char == 'g'))
                            {
                                return Some(6);
                            } else {
                                return None;
                            }
                        }

                        _ => {
                            return None;
                        }
                    }
                }

                // typeof
                't' => {
                    if input == "typeof" {
                        return Some(11);
                    } else {
                        return None;
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // private, finally, extends, package
        7 => {
            match input.as_bytes()[0] as char {
                // extends
                'e' => {
                    if input == "extends" {
                        return Some(29);
                    } else {
                        return None;
                    }
                }

                // finally
                'f' => {
                    if input == "finally" {
                        return Some(16);
                    } else {
                        return None;
                    }
                }

                // private, package
                'p' => {
                    match input.as_bytes()[1] as char {
                        // package
                        'a' => {
                            if input == "package" {
                                return Some(32);
                            } else {
                                return None;
                            }
                        }

                        // private
                        'r' => {
                            if input == "private" {
                                return Some(13);
                            } else {
                                return None;
                            }
                        }

                        _ => {
                            return None;
                        }
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // function, continue
        8 => {
            match input.as_bytes()[0] as char {
                // continue
                'c' => {
                    if input == "continue" {
                        return Some(39);
                    } else {
                        return None;
                    }
                }

                // function
                'f' => {
                    if input == "function" {
                        return Some(35);
                    } else {
                        return None;
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        // interface
        9 => {
            if input == "interface" {
                return Some(34);
            } else {
                return None;
            }
        }

        // instanceof, implements
        10 => {
            match input.as_bytes()[1] as char {
                // implements
                'm' => {
                    if input == "implements" {
                        return Some(33);
                    } else {
                        return None;
                    }
                }

                // instanceof
                'n' => {
                    if input == "instanceof" {
                        return Some(10);
                    } else {
                        return None;
                    }
                }

                _ => {
                    return None;
                }
            }
        }

        _ => {
            return None;
        }
    }
}

File: nested_switch.js
module.exports = function nested_switch(input) {
  switch (input.length) {
    // if, in, do
    case 2:
      switch (input[0]) {
        // do
        case "d":
          if (input[1] == "o") {
            return 40;
          } else {
            return null;
          }

        // if, in
        case "i":
          switch (input[1]) {
            // if
            case "f":
              return 1;

            // in
            case "n":
              return 24;
          }
      }

    // var, get, for, new, any, let, try
    case 3:
      switch (input[0]) {
        // any
        case "a":
          if ((input[1] == "n") & (input[2] == "y")) {
            return 28;
          } else {
            return null;
          }

        // for
        case "f":
          if ((input[1] == "o") & (input[2] == "r")) {
            return 17;
          } else {
            return null;
          }

        // get
        case "g":
          if ((input[1] == "e") & (input[2] == "t")) {
            return 7;
          } else {
            return null;
          }

        // let
        case "l":
          if ((input[1] == "e") & (input[2] == "t")) {
            return 31;
          } else {
            return null;
          }

        // new
        case "n":
          if ((input[1] == "e") & (input[2] == "w")) {
            return 23;
          } else {
            return null;
          }

        // try
        case "t":
          if ((input[1] == "r") & (input[2] == "y")) {
            return 36;
          } else {
            return null;
          }

        // var
        case "v":
          if ((input[1] == "a") & (input[2] == "r")) {
            return 4;
          } else {
            return null;
          }
      }

    // case, else, type, enum, void, null, this, true
    case 4:
      switch (input[0]) {
        // case
        case "c":
          if (input == "case") {
            return 0;
          } else {
            return null;
          }

        // else, enum
        case "e":
          switch (input[1]) {
            // else
            case "l":
              if ((input[2] == "s") & (input[3] == "e")) {
                return 3;
              } else {
                return null;
              }

            // enum
            case "n":
              if ((input[2] == "u") & (input[3] == "m")) {
                return 14;
              } else {
                return null;
              }
          }

        // null
        case "n":
          if (input == "null") {
            return 20;
          } else {
            return null;
          }

        // type, this, true
        case "t":
          switch (input[1]) {
            // this
            case "h":
              if ((input[2] == "i") & (input[3] == "s")) {
                return 22;
              } else {
                return null;
              }

            // true
            case "r":
              if ((input[2] == "u") & (input[3] == "e")) {
                return 26;
              } else {
                return null;
              }

            // type
            case "y":
              if ((input[2] == "p") & (input[3] == "e")) {
                return 9;
              } else {
                return null;
              }
          }

        // void
        case "v":
          if (input == "void") {
            return 19;
          } else {
            return null;
          }
      }

    // throw, while, super, false, yield, const, catch
    case 5:
      switch (input[0]) {
        // const, catch
        case "c":
          switch (input[1]) {
            // catch
            case "a":
              if (input == "catch") {
                return 41;
              } else {
                return null;
              }

            // const
            case "o":
              if (input == "const") {
                return 38;
              } else {
                return null;
              }
          }

        // false
        case "f":
          if (input == "false") {
            return 27;
          } else {
            return null;
          }

        // super
        case "s":
          if (input == "super") {
            return 21;
          } else {
            return null;
          }

        // throw
        case "t":
          if (input == "throw") {
            return 2;
          } else {
            return null;
          }

        // while
        case "w":
          if (input == "while") {
            return 18;
          } else {
            return null;
          }

        // yield
        case "y":
          if (input == "yield") {
            return 37;
          } else {
            return null;
          }
      }

    // number, string, module, typeof, public, export, return, static
    case 6:
      switch (input[0]) {
        // export
        case "e":
          if (input == "export") {
            return 15;
          } else {
            return null;
          }

        // module
        case "m":
          if (input == "module") {
            return 8;
          } else {
            return null;
          }

        // number
        case "n":
          if (input == "number") {
            return 5;
          } else {
            return null;
          }

        // public
        case "p":
          if (input == "public") {
            return 12;
          } else {
            return null;
          }

        // return
        case "r":
          if (input == "return") {
            return 25;
          } else {
            return null;
          }

        // string, static
        case "s":
          switch (input[2]) {
            // static
            case "a":
              if (input == "static") {
                return 30;
              } else {
                return null;
              }

            // string
            case "r":
              if (input == "string") {
                return 6;
              } else {
                return null;
              }
          }

        // typeof
        case "t":
          if (input == "typeof") {
            return 11;
          } else {
            return null;
          }
      }

    // private, finally, extends, package
    case 7:
      switch (input[0]) {
        // extends
        case "e":
          if (input == "extends") {
            return 29;
          } else {
            return null;
          }

        // finally
        case "f":
          if (input == "finally") {
            return 16;
          } else {
            return null;
          }

        // private, package
        case "p":
          switch (input[1]) {
            // package
            case "a":
              if (input == "package") {
                return 32;
              } else {
                return null;
              }

            // private
            case "r":
              if (input == "private") {
                return 13;
              } else {
                return null;
              }
          }
      }

    // function, continue
    case 8:
      switch (input[0]) {
        // continue
        case "c":
          if (input == "continue") {
            return 39;
          } else {
            return null;
          }

        // function
        case "f":
          if (input == "function") {
            return 35;
          } else {
            return null;
          }
      }

    // interface
    case 9:
      if (input == "interface") {
        return 34;
      } else {
        return null;
      }

    // instanceof, implements
    case 10:
      switch (input[1]) {
        // implements
        case "m":
          if (input == "implements") {
            return 33;
          } else {
            return null;
          }

        // instanceof
        case "n":
          if (input == "instanceof") {
            return 10;
          } else {
            return null;
          }
      }
  }

  return null;
};
